File: BungiiAutomation/Bungii.Test.Regression.Android.Integration/Pages/MenuPages/SaveMoneyPage.cs
using OpenQA.Selenium;
using OpenQA.Selenium.Support.PageObjects;

namespace Bungii.Test.Regression.Android.Integration.Pages
{
    class SaveMoneyPage
    {
        public SaveMoneyPage(IWebDriver driver)
        {
           PageFactory.InitElements(driver, this);
        }

        [FindsBy(How = How.XPath, Using = "//android.widget.TextView[@text='SAVE MONEY']")]     
        public IWebElement Header_SavePage { get; set; }   //places_autocomplete_pickup_location is actual id 

        [FindsBy(How = How.Id, Using = "com.bungii.customer:id/field_promo_code")]
        public IWebElement Textfield_PromoCode { get; set; }

        [FindsBy(How = How.XPath, Using = "//TextInputLayout[@resource-id='com.bungii.customer:id/field_promo_code_layout']/following-sibling::android.widget.Button")]
        public IWebElement Button_Add { get; set; }

        [FindsBy(How = How.Id, Using = "com.bungii.customer:id/btn_get_more_money")]
        public IWebElement Button_GetMoreMoney { get; set; }

        [FindsBy(How = How.XPath, Using = "//android.widget.TextView[@resource-id='com.bungii.customer:id/promo_code_label'][1]")]
        public IWebElement SaveMoney_PromoCode1 { get; set; }

        [FindsBy(How = How.XPath, Using = "//android.widget.TextView[@resource-id='com.bungii.customer:id/promo_code_label'][2]")]
        public IWebElement SaveMoney_PromoCode2 { get; set; }

        [FindsBy(How = How.XPath, Using = "//android.widget.TextView[@resource-id='com.bungii.customer:id/promo_code_label'][3]")]
        public IWebElement SaveMoney_PromoCode3 { get; set; }
    }
}

File: Bungii.Test.Integration/src/test/java/com/bungii/android/pages/menus/SaveMoneyPage.java
package com.bungii.android.pages.menus;

import com.bungii.common.core.PageBase;
import org.openqa.selenium.WebElement;

public class SaveMoneyPage extends PageBase {
    public WebElement Header_SavePage () { return findElement("//android.widget.TextView[@text='SAVE MONEY']", LocatorType.XPath); }   //places_autocomplete_pickup_location is actual id

    public WebElement Textfield_PromoCode () { return findElement("com.bungii.customer:id/field_promo_code", LocatorType.Id); }

    public WebElement Button_Add () { return findElement("//TextInputLayout[@resource-id='com.bungii.customer:id/field_promo_code_layout']/following-sibling::android.widget.Button", LocatorType.XPath); }


    public WebElement Button_AddPromoPage () { return findElement("android.widget.Button", LocatorType.ClassName); }

    public WebElement Button_GetMoreMoney () { return findElement("com.bungii.customer:id/btn_get_more_money", LocatorType.Id); }

    public WebElement SaveMoney_PromoCode1 () { return findElement("//android.widget.TextView[@resource-id='com.bungii.customer:id/promo_code_label'][1]", LocatorType.XPath); }

    public WebElement SaveMoney_PromoCode2 () { return findElement("//android.widget.TextView[@resource-id='com.bungii.customer:id/promo_code_label'][2]", LocatorType.XPath); }

    public WebElement SaveMoney_PromoCode3 () { return findElement("//android.widget.TextView[@resource-id='com.bungii.customer:id/promo_code_label'][3]", LocatorType.XPath); }
}
